board.go

```diff
@@ -164,3 +181,24 @@ func (s *BoardService) GetAllSprints(boardID string) ([]Sprint, *Response, error
 
 	return result.Sprints, resp, err
 }
+
+// GetAllSprintsWithOptions will return sprints from a board, for a given board Id and filtering options
+// This only includes sprints that the user has permission to view.
+//
+// JIRA API docs: https://docs.atlassian.com/jira-software/REST/cloud/#agile/1.0/board/{boardId}/sprint
+func (s *BoardService) GetAllSprintsWithOptions(boardID int, options *GetAllSprintsOptions) (*SprintsList, *Response, error) {
+	apiEndpoint := fmt.Sprintf("rest/agile/1.0/board/%d/sprint", boardID)
+	url, err := addOptions(apiEndpoint, options)
+	req, err := s.client.NewRequest("GET", url, nil)
```

Comment:
Also, what do you think about having `GetAllSprints` call this with nil options.  That should avoid some repeated code.

Reply:
Yeah, that's a much better idea. I shouldn't be rushing :)